Examples/Algorithms/Geant4HepMC/src/EventAction.cpp

```diff
@@ -181,7 +218,8 @@ void ActsExamples::EventAction::EndOfEventAction(const G4Event*) {
   save_event(&m_event,"",__LINE__);
   // Filter irrelevant processes
   auto currentVertex = m_event.vertices()[0];
-  followOutgoingParticles(m_event, currentVertex, m_processFilter);
+//  followOutgoingParticles(m_event, currentVertex, m_processFilter);
+  followOutgoingParticlesTrivialAncestor(m_event, m_processFilter);
```

Comment:
This actually replaces the old function, right?

Reply:
Yes and no. It could make sense to keep both algorithms for comparison.

Reply:
You mean semantic comparison?

Reply:
I mean running  `diff` on the output produced with the first and second algorithms. A kind of cross-check that these produce identical results.

Reply:
This is what I had in mind. If the results are the same, then the old one can be removed. Due to this PR the old function would be dead code anyway. Do you think an `operator==` in `GenEvent` would be reasonable? If such a function would exist then both could be compared once in this line here.

Reply:
In the  'If the results will be the same' 'if' is the most important part. But even if one test now shows the outputs are equivalent, one of the two algorithms would be not a 'dead code'  but a cross-check.  Yes, I know that ACTS runs code coverage tests. To avoid complaints one could actually put a flag and have both implementations.

On the comparison: '==' requires some event to compare to, i.e. some reference. If one keeps both algorithms and checks one output vs. another an actual `diff` is the most reasonable solution. Literally a `cmake` test that runs `diff A.txt B.txt`
Also, a self-made diff could be made super-portable.

Reply:
Ok, I've implemented a simple switch which is initialized as a constant now and has no settters/getters.

Reply:
Nice, thanks :)